### PR TITLE
test: code coverage 70% Split Expense Page

### PR DESCRIPTION
**Pull request by CODEX108**
### <samp>🤖 Generated by Copilot at ca47a6f</samp>

This pull request updates and adds mock data files for various features in the app, such as flattened accounts, file objects, recently used items, and split expenses. It also improves the type safety and readability of some models and components that use the mock data. It fixes some property names and values and imports the correct mock data arrays for testing.

### <samp>🤖 Generated by Copilot at ca47a6f</samp>

> _Mock data changes_
> _`OrgCategory` and `File` -_
> _Autumn tests await_

### <samp>🤖 Generated by Copilot at ca47a6f</samp>

*  Import `FlattenedAccount` model and add type annotations to `flattenedData` object in mock data ([link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-62f534a3ab6da273bee558942f614746004cbcaf0b2a8ea5b22342a27deff29fR1-R2), [link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-62f534a3ab6da273bee558942f614746004cbcaf0b2a8ea5b22342a27deff29fL133-R135))
*  Rename and reformat `advance_number1` and `number1` properties in mock data to match `FlattenedAccount` model and backend response ([link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-62f534a3ab6da273bee558942f614746004cbcaf0b2a8ea5b22342a27deff29fL151-R153), [link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-62f534a3ab6da273bee558942f614746004cbcaf0b2a8ea5b22342a27deff29fL186-R190))
*  Add new mock data object `fileObjectData5` with `receipt_coordinates` property for testing receipt cropping feature ([link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-b5604afda033e8ee07244ac5848af54d72cfd93a0fefa253923914e0a68346c8R68-R80))
*  Add new mock data arrays `fileObject6`, `fileObject7`, and `fileObject8` with different combinations of file objects for testing file attachment feature ([link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-b5604afda033e8ee07244ac5848af54d72cfd93a0fefa253923914e0a68346c8R223-R291))
*  Add new mock data file `org-category-list-item.data.ts` with two arrays of type `OrgCategoryListItem[]` for testing category selection feature ([link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-3f08ed1afe1b627dd8d79ee0eb6e1b895d7e1505e3670f46c43b97c46e016c9eR1-R69))
*  Remove unused import and array from `recently-used.data.ts` after moving `recentUsedCategoriesRes` array to `org-category-list-item.data.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-6512bbe410b564ad644cd215bf5f483358a11ae57e943d817a0da542b98d8cf3L5), [link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-6512bbe410b564ad644cd215bf5f483358a11ae57e943d817a0da542b98d8cf3L191-L238))
*  Import `dayjs` library and add new mock data objects `splitExpense3` to `splitExpense7` with different values for testing split expense feature ([link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-a4a7e89114db7ccca42a132a450ff980137b8499ece1f6b2da02c4d89234f1f0R1), [link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-a4a7e89114db7ccca42a132a450ff980137b8499ece1f6b2da02c4d89234f1f0R48-R86))
*  Modify `number` property of `advance` object in `ExtendedAccount` interface to accept either string or number type ([link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-32bd164da0070ed37bbcdd153d59df06277592cdb9efb1b1b8c275895c55c381L14-R14))
*  Modify `files` property of `FileTransaction` interface to be optional ([link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-7a3d072518be859cc7f9c790b69a0a69820ed1aa2a92063fab54ae124cbf0bf0L5-R5))
*  Modify import of `recentUsedCategoriesRes` array in `recently-used-items.service.spec.ts` to reflect new location ([link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-d3dccc1dc715b9b625b7f0022b99d7bb029cd9926e986e89bef824aad9b72fb1L14-R18))
*  Import `OrgCategoryListItem` and `FileObject` models and modify type annotations of `categories$` and `fileObjs` in `split-expense.page.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L23-R23), [link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878R31), [link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L56-R57), [link](https://github.com/fylein/fyle-mobile-app/pull/2148/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L64-R65))

## Clickup
https://app.clickup.com/t/85zt9c323

## Code Coverage
Please add code coverage here

image

### CODE REVIEW

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
discussed on slack

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Resolve conflicts

File: src/app/core/mock-data/file-txn.data.ts [hunk 1]
```diff
@@ -406,7 +412,7 @@ export const fileTxns2: FileTransaction = {
 };
 
 export const fileTxns3: FileTransaction = {
-  txns: splitExpenseTxn1,
+  txns: [{ ...splitExpenseTxn1 }, { ...splitExpenseTxn1_1, id: 'tx12SqYytrm' }],
```
Comment (yksolanki9) [inline src/app/core/mock-data/file-txn.data.ts:415]: `{ ...splitExpenseTxn1 }`  - why do we need the spread operator here?

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 2]
```diff
@@ -264,7 +289,82 @@ fdescribe('SplitExpensePage', () => {
...
+        txn_dt: new FormControl('2023-01-11'),
+        category: new FormControl(''),
+      });
+      Object.defineProperty(splitExpenseForm1.controls.amount, '_pendingChange', { value: true });
+      component.splitExpensesFormArray = new FormArray([splitExpenseForm1, otherSplitExpenseForm]);
+
+      component.onChangeAmount(splitExpenseForm1, 0);
+      tick(500);
+
+      expect(splitExpenseForm1.controls.amount.value).toEqual(120);
```
Comment (yksolanki9) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:317]: Shouldnt you be checking if the values of the other form are getting updated?

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 3]
```diff
@@ -264,7 +289,82 @@ fdescribe('SplitExpensePage', () => {
...
+      expect(splitExpenseForm1.controls.percentage.value).toEqual(6);
+
+      component.onChangeAmount(otherSplitExpenseForm, 1);
+      tick(500);
+      expect(component.splitExpensesFormArray.controls[1].value.amount).toEqual(1880);
```
Comment (yksolanki9) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:322]: I'm confused what you're trying to check here. You're setting the amount for form at index 1 to 800 and then asserting it against 1880?

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 4]
```diff
@@ -264,7 +289,82 @@ fdescribe('SplitExpensePage', () => {
...
+      expect(otherSplitExpenseForm.controls.percentage.value).toEqual(4);
+
+      component.onChangeAmount(splitExpenseForm1, 0);
+      tick(500);
+      expect(component.splitExpensesFormArray.controls[0].value.amount).toEqual(1920);
+      expect(component.splitExpensesFormArray.controls[0].value.percentage).toEqual(96);
+      expect(splitExpenseForm1.controls.amount.value).toEqual(1920);
+      expect(splitExpenseForm1.controls.percentage.value).toEqual(96);
+      expect(otherSplitExpenseForm.controls.percentage.value).toEqual(4);
+      expect(component.getTotalSplitAmount).toHaveBeenCalledTimes(1);
```
Comment (yksolanki9) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:363]: Same comments here

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 5]
```diff
@@ -895,6 +917,157 @@ fdescribe('SplitExpensePage', () => {
...
+      const expectedMinDate = `${minDate.getFullYear()}-${minDate.getMonth() + 1}-${minDate.getDate()}`;
+      const expectedMaxDate = `${maxDate.getFullYear()}-${maxDate.getMonth() + 1}-${maxDate.getDate()}`;
+
+      component.setValuesForCCC(currencyObj, homeCurrency, isCorporateCardsEnabled);
+      expect(component.setAmountAndCurrency).toHaveBeenCalledWith(currencyObj, homeCurrency);
+      expect(component.add).toHaveBeenCalledWith(amount1, 'INR', percentage1, null);
+      expect(component.add).toHaveBeenCalledWith(amount2, 'INR', percentage2, null);
+      expect(component.getTotalSplitAmount).toHaveBeenCalledTimes(3);
+      expect(dateService.addDaysToDate).toHaveBeenCalledTimes(1);
+      expect(component.minDate).toEqual(expectedMinDate);
```
Comment (yksolanki9) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:1031]: maxDate?

**Review by yksolanki9 — CHANGES_REQUESTED** [review]

File: src/app/fyle/split-expense/split-expense.page.spec.ts (hunk 3 above)
Comment (CODEX108) [reply src/app/fyle/split-expense/split-expense.page.spec.ts]: I've tried resolving this by rearranging the order of assertions

> You're setting the amount for form at index 1 to 800 and then asserting it against 1880?
Actually the amount value in the beginning is 800 and after the method is called this value is updated and changed to 1880. (i.e the form2 value) , along with the percentage

Then the percentage value of form1 is changed.

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 6]
```diff
@@ -313,16 +303,13 @@ describe('SplitExpensePage', () => {
 
       component.onChangeAmount(splitExpenseForm1, 0);
       tick(500);
-
-      expect(splitExpenseForm1.controls.amount.value).toEqual(120);
-      expect(splitExpenseForm1.controls.percentage.value).toEqual(6);
+      expect(otherSplitExpenseForm.controls.amount.value).toEqual(1880);
+      expect(otherSplitExpenseForm.controls.percentage.value).toEqual(94);
 
       component.onChangeAmount(otherSplitExpenseForm, 1);
```
Comment (yksolanki9) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:309]: Why is this needed? You're already setting the value to 120, so changing the amount of other expense should not matter

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 7]
```diff
@@ -350,15 +337,12 @@ describe('SplitExpensePage', () => {
...
       tick(500);
 
-      expect(otherSplitExpenseForm.controls.amount.value).toEqual(80);
-      expect(otherSplitExpenseForm.controls.percentage.value).toEqual(4);
+      expect(splitExpenseForm1.controls.amount.value).toEqual(1920);
+      expect(splitExpenseForm1.controls.percentage.value).toEqual(96);
 
       component.onChangeAmount(splitExpenseForm1, 0);
```
Comment (yksolanki9) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:343]: Same comment here

File: src/app/fyle/split-expense/split-expense.page.spec.ts [hunk 8]
```diff
@@ -997,7 +981,12 @@ describe('SplitExpensePage', () => {
 
   describe('setValuesForCCC():', () => {
     it('should set the values for CCC split expenses when coporate credit cards is enabled', () => {
-      dateService.addDaysToDate.and.returnValue(new Date());
+      const today = new Date();
```
Comment (yksolanki9) [inline src/app/fyle/split-expense/split-expense.page.spec.ts:984]: Didnt understand this change, what was wrong with the previous approach?

**Review by yksolanki9 — CHANGES_REQUESTED** [review]

**Review by Dimple16 — COMMENTED** [review]
my review is not required

File: src/app/fyle/split-expense/split-expense.page.spec.ts (hunk 5 above)
Comment (yksolanki9) [reply src/app/fyle/split-expense/split-expense.page.spec.ts:1012]: https://day.js.org/docs/en/manipulate/add

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Discussed on call

**Review by yksolanki9 — APPROVED** [review]

**Pull request merged**